cmd/user-agent/src/pages/chapi/wallet/get/getCredentialsByQuery.js

```diff
@@ -110,3 +124,88 @@ function retainOnlySelected(presentationSubmission, selectedIndexes){
 
     return presentationSubmission
 }
+
+async function getConsentCredentials(aries, presentationSubmission, walletManager) {
+    let vcs = []
+    // TODO DID exchange with verifier not needed for now
+    // let exchange = new DIDExchange(aries)
+    // let connection = await exchange.connect(invitation)
+    // console.log(`got connection with ${connection.result.State} status`, connection)
+
+    //TODO to improve performance all request should be sent to all issuers in paralle and need to establish
+    // cooreltation between incoming actions
+    for (let vc of presentationSubmission.verifiableCredential) {
+        if (getCredentialType(vc.type) != 'IssuerManifestCredential') {
+            vcs.push(vc)
+            return
+        }
+
+        let connection = await walletManager.getConnectionByID(vc.connection)
+
+        //TODO pass user DID & RP DID as ~attachments
+        aries.issuecredential.sendRequest({
+            my_did: connection.MyDID,
+            their_did: connection.TheirDID,
+            request_credential: {},
+        })
+
+        let event = await waitForNotification(aries, ["issue-credential_states"], "post_state")
+        if (event.StateID != 'request-sent') {
+            throw 'failed to send credential request to issuer'
+        }
+
+        console.log("sent credential request state notification", event.Message['@id'])
+
+        let piid = event.Message['@id']
+        let action = await getAction(aries, piid)
+        aries.issuecredential.acceptCredential({
+            piid: action.PIID,
+            names: [consentCredentialType],
+        })
+
+        let credential = await getCredential(aries, consentCredentialType)
+        vcs.push(credential)
+    }
+
+    presentationSubmission.verifiableCredential = vcs
+    return presentationSubmission
+}
+
+const retries = 15;
```

Comment:
this can take up to 15 seconds!

Reply:
Whole logic is going to be replaced, based on TODO above.